src/gallium/state_trackers/nine/texture9.c

```diff
@@ -337,9 +337,13 @@ NineTexture9_AddDirtyRect( struct NineTexture9 *This,
         u_box_origin_2d(This->base.base.info.width0,
                         This->base.base.info.height0, &This->dirty_rect);
     } else {
-        struct pipe_box box;
-        rect_to_pipe_box(&box, pDirtyRect);
-        u_box_union_2d(&This->dirty_rect, &This->dirty_rect, &box);
+        if (This->dirty_rect.width == 0) {
+            rect_to_pipe_box(&This->dirty_rect, pDirtyRect);
+        } else {
+            struct pipe_box box;
+            rect_to_pipe_box(&box, pDirtyRect);
+            u_box_union_2d(&This->dirty_rect, &This->dirty_rect, &box);
+        }
         if (u_box_clip_2d(&This->dirty_rect, &This->dirty_rect,
```

Comment:
Interesting fix !

However nine makes other usages of u_box_union_*d which needs to be fixed as well.
Perhaps u_box_union_*d could have a special handling for empty boxes as well... Not sure what is best.
Nine is the only user of the functions, so no problem to modify them.